### ISSUE TITLE
Astro not respect config base path + not resolve build files correctly on base path

### ISSUE THREAD

**Issue by harrytran998** [withastro/astro#5413]
### What version of `astro` are you using?

1.6.8

### Are you using an SSR adapter? If so, which one?

node

### What package manager are you using?

yarn 3

### What operating system are you using?

Mac

### Describe the Bug

Here the astro config

```js
export default defineConfig({
  site,
  output: 'server',
  base: '/ok',
  adapter: AstroNode({
    mode: 'standalone'
  }),
  ```

<img width="638" alt="image" src="https://user-images.githubusercontent.com/24296018/202087090-77527580-5452-46d7-96a7-25d2090af2c1.png">

When I config the base path in Astro, I expect this config will transform the base path to `/ok` instead of `/`.  Hopefully, your team takes a look. This is really important with products having multiple pages, and for some pages, we want to use Astro --> Need to use base config.

### Link to Minimal Reproducible Example

https://github.com/harrytran998/reproduction-astro-base-path

### Participation

- [ ] I am willing to submit a pull request for this issue.

**Comment by surjithctly** [comment]
Docs says, it has no effect on developement. It will work fine on build though.
https://docs.astro.build/en/reference/configuration-reference/#base
![image](https://user-images.githubusercontent.com/1884712/202176483-e8f3d48f-15bf-4fd3-be84-61e757218354.png)

**Comment by harrytran998** [comment]
Sorry about this, but in production mode --> 
`http://0.0.0.0:3000/ok` Not work
But `http://0.0.0.0:3000/ok/` work 🥲

**Comment by harrytran998** [comment]
And the assets files seem not to work locally When I build with UnoCss, this generates some asset files like CSS --> But seems Astro cannot resolve this.

`http://0.0.0.0:3000/ok/assets/index.5cfd13f6.css`

**Comment by matthewp** [comment]
Looks like this is specifically with the Node adapter, right? I'll take a look and see what i can figure out here.

**Comment by matthewp** [comment]
Oh, this was fixed in `@astrojs/node` version 3.0.0. If you upgrade you'll see this working.
